lib/livebook_web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,20 @@
+defmodule LivebookWeb.AuthController do
+  use LivebookWeb, :controller
+
+  alias LivebookWeb.Helpers
+
+  def index(conn, assigns) do
```

Comment:
We should not be able to access this action if we are already authenticate. My suggestion is to break `LivebookWeb.Auth` to be three distinct functions:

1. authenticate
2. ensure_authenticated
3. require_no_authentication

authenticate is simply going to set an assign saying if the user is authenticated or not. then ensure_authenticated will redirect if the user is not signed in and require_no_authentication, to be used by this controller, redirects if the user *is* signed in.

Reply:
I don't know if I understand that correctly. What you meant by that is we should not trigger error page when somebody is authorized and happened to access `GET /autenticate'  therefore we should check if the user is authorized and redirect him accordingly?

Reply:
Correct!